tutorial/LFRic/example1/Makefile

```diff
@@ -0,0 +1,101 @@
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ------------------------------------------------------------------------------
+# Authors: I. Kavcic, Met Office and J. Henrichs, Bureau of Meteorology
+
+# Compiler settings
+F90 ?= gfortran
+F90FLAGS ?= -Wall -g -fcheck=bound
+
+# Source names (without extensions)
+DRIVER = example1_driver
+ALGORITHM = example1_alg_mod
+PSY = $(ALGORITHM:=_psy)
+KERNEL = setval_field_w0_kernel_mod setval_field_w3_kernel_mod
+
+# PSyclone command-line options
+PSYCLONE_CMD = -nodm -l all
+
+# Executable
+EXEC = example1.out
+
+# LFRic infrastructure library
+PSYCLONE_RELPATH = ../../..
+LFRIC_PATH = $(PSYCLONE_RELPATH)/src/psyclone/tests/test_files/dynamo0p3/infrastructure
+LFRIC_NAME=lfric
+LFRIC_LIB=$(LFRIC_PATH)/lib$(LFRIC_NAME).a
+F90FLAGS += -I$(LFRIC_PATH)
+
+# Source and object rules
+DRIVER_SRC = $(DRIVER:=.f90)
+ALGORITHM_SRC = $(ALGORITHM:=.f90)
+PSY_SRC = $(PSY:=.f90)
+KERNEL_SRC = $(KERNEL:=.f90)
+
+DRIVER_OBJ = $(filter %.o,$(DRIVER_SRC:.f90=.o))
+ALGORITHM_OBJ = $(filter %.o,$(ALGORITHM_SRC:.f90=.o))
+PSY_OBJ = $(filter %.o,$(PSY_SRC:.f90=.o))
+KERNEL_OBJ = $(filter %.o,$(KERNEL_SRC:.f90=.o))
+EXAMPLE_OBJ = $(DRIVER_OBJ) $(ALGORITHM_OBJ) $(PSY_OBJ) $(KERNEL_OBJ)
+
+default: $(EXEC)
+
+$(EXEC): $(LFRIC_LIB) $(EXAMPLE_OBJ)
+	$(F90) $(F90FLAGS) $(EXAMPLE_OBJ) -o $(EXEC) -L$(LFRIC_PATH) -l$(LFRIC_NAME)
+
+$(LFRIC_LIB):
+	$(MAKE) -C $(LFRIC_PATH)
+
+# Dependencies
+$(DRIVER_OBJ): $(ALGORITHM_OBJ) $(PSY_OBJ) $(KERNEL_OBJ)
+$(ALGORITHM_OBJ): $(PSY_OBJ) $(KERNEL_OBJ)
+$(PSY_OBJ): $(KERNEL_OBJ)
+
+%.o %.mod: %.F90
+	$(F90) $(F90FLAGS) -c $<
+
+%.o %.mod: %.f90
+	$(F90) $(F90FLAGS) -c $<
+
+# Keep the generated Alg and PSy files
+.precious: $(ALGORITHM_SRC) $(PSY_SRC)
+
+$(ALGORITHM_SRC): $(PSY_SRC)
+
+%_psy.f90: %.x90
+	psyclone $(PSYCLONE_CMD) --config $(PSYCLONE_RELPATH)/config/psyclone.cfg \
+	-opsy $*_psy.f90 -oalg $*.f90 $<
+
+clean:
+	rm -f *.o *.mod $(LFRIC_PATH)/*.o $(LFRIC_PATH)/*.mod $(LFRIC_PATH)/*.a \
```

Comment:
If any of these files don't exist `rm` may return an error causing the Makefile to fail. You can prepend the command with `-` to have Make ignore the error. i.e. `-rm ...`

Reply:
`rm` without `-` actually works fine when a file does not exist, so I left them as they are.